src/rs/lib.rs

```diff
@@ -1479,11 +1596,7 @@ mod tests {
         );
 
         let mut connection = Connection::new();
-        let res = connection.open(
-            &registration,
-            Some(test_conn_callback),
-            &connection as *const Connection as *const c_void,
-        );
+        let res = connection.open(&registration, test_conn_callback);
```

Comment:
It would be interesting to keep a test that provide some context (since any non-trivial application would need to), to make sure it is correctly handled through Rust lifetime validation + unsafe code. A counter for the number of calls could be an easy one.

Reply:
I have a full server client test in another branch. Will make it to main soon.

Reply:
Added the full server client test in this PR.